js/compressed-token/src/instructions/transfer.ts

```diff
@@ -1,101 +1,115 @@
 import {
-  CompressedProof_IdlType,
+  CompressedProof,
   defaultStaticAccountsStruct,
-  InUtxoTuple_IdlType,
+  CompressedAccountWithMerkleContext,
   LightSystemProgram,
-  Utxo_IdlType,
+  PackedCompressedAccountWithMerkleContext,
 } from '@lightprotocol/stateless.js';
 import {
   PublicKey,
   TransactionInstruction,
   AccountMeta,
+  ComputeBudgetProgram,
 } from '@solana/web3.js';
 import { CompressedTokenProgram } from '../program';
-import { TokenTlvData_IdlType, TokenTransferOutUtxo_IdlType } from '../types';
+import {
+  CompressedTokenInstructionDataTransfer,
+  TokenData,
+  TokenTransferOutputData,
+} from '../types';
 
-// NOTE: this is currently akin to createExecuteCompressedInstruction on-chain
+/// TODO: refactor akin to lightsystemprogram.transfer()
 export async function createTransferInstruction(
   feePayer: PublicKey,
   authority: PublicKey,
-  inUtxoMerkleTreePubkeys: PublicKey[],
-  nullifierArrayPubkeys: PublicKey[],
-  outUtxoMerkleTreePubkeys: PublicKey[],
-  inUtxos: Utxo_IdlType[],
-  outUtxos: TokenTransferOutUtxo_IdlType[], // tlv missing
-  rootIndices: number[],
-  proof: CompressedProof_IdlType,
-): Promise<TransactionInstruction> {
-  const outputUtxos = outUtxos.map((utxo) => ({ ...utxo }));
+  inputStateTrees: PublicKey[],
+  inputNullifierQueues: PublicKey[],
+  outputStateTrees: PublicKey[],
+  inputCompressedAccounts: CompressedAccountWithMerkleContext[],
+  outputCompressedAccounts: TokenTransferOutputData[],
+  recentStateRootIndices: number[],
+  recentValidityproof: CompressedProof,
+): Promise<TransactionInstruction[]> {
   const remainingAccountsMap = new Map<PublicKey, number>();
-  const inUtxosWithIndex: InUtxoTuple_IdlType[] = [];
-  const inUtxoTlvData: TokenTlvData_IdlType[] = [];
+  const packedInputCompressedAccountsWithMerkleContext: PackedCompressedAccountWithMerkleContext[] =
+    [];
+  const inputTokenData: TokenData[] = [];
 
   const coder = CompressedTokenProgram.program.coder;
 
-  inUtxoMerkleTreePubkeys.forEach((mt, i) => {
+  /// packs, also sets data zero, so if they have data they need to extract it beforehand
```

Comment:
maybe this should throw an error if there is data instead of silently setting it to zero or at least print a warning when it is setting data zero.

Reply:
comment was confusing, fixed